opcodes/arc-opc.c

```diff
@@ -1695,7 +1695,7 @@ const struct arc_operand arc_operands[] =
 
   /* SIMM12_20 mask = 00000000000000000000111111222222.  */
 #define SIMM12_20	(UIMM6_20 + 1)
-  {12, 0, 0, ARC_OPERAND_SIGNED, insert_simm12_20, extract_simm12_20},
+  {12, 0, -SIMM12_20, ARC_OPERAND_SIGNED, insert_simm12_20, extract_simm12_20},
```

Comment:
We hit an backend limitation, which I will need to address it latter on. Namely, the selection of the right opcode when the mnemonic demands a relocation. In our case we have a reloc for s12 and a reloc for limm. The assembler picks the first one i.e. s12 reloc. Hence, the issue.
Anyhow, I did fixed for u, and as far as I can see there shouldn't be any error. You can find the patch here:
https://github.com/foss-for-synopsys-dwc-arc-processors/binutils-gdb/tree/jli